pkg/apis/triggers/v1alpha1/trigger_validation.go

```diff
@@ -70,21 +70,25 @@ func (t TriggerSpecTemplate) validate(ctx context.Context) *apis.FieldError {
 }
 
 func (t triggerSpecBindingArray) validate(ctx context.Context) *apis.FieldError {
-	if len(t) > 0 {
-		for i, b := range t {
-			// Either Ref or Spec should be present
-			if b.Ref == "" && b.Spec == nil {
-				return apis.ErrMissingOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			// Both Ref and Spec can't be present at the same time
-			if b.Ref != "" && b.Spec != nil {
+	for i, b := range t {
+		switch {
+		case b.Ref != "":
+			switch {
+			case b.Spec != nil: // Cannot specify both Ref and Spec
 				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Spec", i))
-			}
-
-			if b.Ref != "" && b.Kind != NamespacedTriggerBindingKind && b.Kind != ClusterTriggerBindingKind {
+			case b.Name != "": // Cannot specify both Ref and Name
+				return apis.ErrMultipleOneOf(fmt.Sprintf("bindings[%d].Ref", i), fmt.Sprintf("bindings[%d].Name", i))
+			case b.Kind != NamespacedTriggerBindingKind && b.Kind != ClusterTriggerBindingKind: // Kind must be valid
```

Comment:
When these conditions are applicable :thinking:

Reply:
if we use Ref, we cannot also provide a Name. If we use Ref, we can optionally provide a Kind whose valid values are TriggerBinding or ClusterTriggerBinding. Does that answer your question? Or were you looking for something else?